### ISSUE TITLE
[Native] The "Extra Credentials" API is not available to Prestissimo/Velox connectors

### ISSUE THREAD

**Issue by ashkrisk** [prestodb/presto#22849]
As per the [Presto Client REST API](https://prestodb.io/docs/current/develop/client-protocol.html#client-request-headers), clients can add a header called `X-Presto-Extra-Credential` to their requests. These extra credentials are not used by core Presto, but are made available to the connectors through the [`ConnectorSession`](https://github.com/prestodb/presto/blob/831d5947b909fee0d5c0091a3246ddc5b31b2731/presto-spi/src/main/java/com/facebook/presto/spi/ConnectorSession.java#L28) object.
```java
session.getIdentity().getExtraCredentials()
```
In Prestissimo, the closest equivalent to the `ConnectorSession` is the [`ConnectorQueryCtx`](https://github.com/facebookincubator/velox/blob/cbf9b55cf61d23016e3a5a918c7ac4a8c18faf45/velox/connectors/Connector.h#L254) class, which does not have any interface to read the extra credentials provided by the client.

As such there is no way for Prestissimo connectors to read the extra credentials supplied by the client.

## Expected Behavior or Use Case
Connectors written for Prestissimo should have access to the extra credentials provided by the client.

## Presto Component, Service, or Connector
Prestissimo connector infrastructure.

## Context
Normally, access credentials are specified as catalog properties. This does not always work well in multi-user contexts, where many different users with varying levels of access credentials are all interacting with the same presto cluster. The "Extra Credentials" interface is a way for integrations to provide different levels of access to different users, but these are currently unavailable to Prestissimo connectors. We ran into this issue while developing an [Arrow Flight connector](https://github.com/prestodb/rfcs/pull/10) for Prestissimo.

**Comment by aditi-pandit** [comment]
@majetideepak

**Comment by ashkrisk** [comment]
@majetideepak yes, we can work with this, thanks 👍
Should the credentials added to the session be prefixed like `ec.<name>` or `extraCreds.<name>` to avoid conflicts with the actual session properties?

**Comment by tdcmeehan** [comment]
Wondering if this use case, providing per-session authentication credentials, is common enough that it should have its own abstraction in Velox?

**Comment by majetideepak** [comment]
> Wondering if this use case, providing per-session authentication credentials, is common enough that it should have its own abstraction in Velox?

Yes, we should add an identity/credentials abstraction in Velox. The complexity is to make this engine agnostic.

**Comment by tdcmeehan** [comment]
> > Wondering if this use case, providing per-session authentication credentials, is common enough that it should have its own abstraction in Velox?
> 
> Yes, we should add an identity/credentials abstraction in Velox. The complexity is to make this engine agnostic. Is it going to simply be another map?

That's how it is in Presto.  Connectors will either expect a specific key in that map, or it may be configurable as a property.

**Comment by majetideepak** [comment]
I added a simple `Identity` API in Velox. Let me know your feedback.
https://github.com/facebookincubator/velox/pull/9982

**Comment by majetideepak** [comment]
@ashkrisk can you update the RFC to explain how and where the extraCredentials will be used?

**Comment by ashkrisk** [comment]
@majetideepak added a couple of lines to explain where the extraCredentials will be used.

https://github.com/prestodb/rfcs/blob/1c8a264658eda3c47e2ba4bda2ecb32fb391992f/RFC-0003-arrow-flight-connector.md?plain=1#L110-L112
